DOCS/man/options.rst

```diff
@@ -1027,6 +1027,11 @@ Program Behavior
     Enable the builtin script that lets you select from lists of items (default:
     yes). By default, its keybindings start with the ``g`` key.
 
+``--load-autoload=<yes|no>``
+    Enable the builtin script that scans the directory a file is located in and
+    adds the found files to the playlist before and after the currently playing
+    file (default: no).
```

Comment:
I think all builtin scripts should be loaded by default, else there is no point in bundling them.

Reply:
I would strongly oppose loading this by default since the behavior, imo makes no sense if you are using a cli. The argument for including this is that it's a reasonably commonly requested feature that some users like and `autoload` for all intents and purposes is a first-party script anyway. It's easier for them to set a line in the config file as opposed to copying a file in their script folder, tracking upstream changes/fixes, etc.

Reply:
lol, makes sense

Reply:
I agree that in this case it would be useful to make it a builtin, but keep it disabled by default.

However, I find the new docs rather sparse, though I don't have in mind a better version (and I don't use autoload myself).

Reply:
> I would strongly oppose loading this by default since the behavior

Reply:
> I would strongly oppose loading this by default since the behavior

I'm not saying it should be made default behavior. But up until now all bundled scripts are loaded by default and their behavior is controlled by `script-opts`. Bundling the script to not load it is fine, but one can argue if it is really that helpful.

Either way, if we want to control this script by loading and unloading it, I would give it more descriptive name like `--autocreate-playlist` or something that ties to functionality and not implementation detail of `load` script.

And `autoload.rst` need to be added with documentation of this script, similar to others.

> autoload is miles away the most popular one.

We don't know that. Most people doesn't even know about it and the others doesn't want playlists to be created for them if they play single file. But I agree it can be bundled, same as (I know meme) autocrop could :)

Reply:
You should enable the script by default.
You just need to make its first option's default value to `yes` instead.
https://github.com/mpv-player/mpv/blob/5c6caa29c9a7099e6f0034488df26953bf21efe6/player/lua/autoload.lua#L43

If disable the script by default. What's the meaning of this option?

Reply:
I left out documenting the options partially because I'm lazy but also because I wasn't sure where to put it. Should it be a separate `rst` or just within options like `ytdl_hook` is? As for the naming, I dunno but it should have 'autoload' in it at least. `--autoload-playlist`?

Reply:
`autoload` is misleading name for me. We are not "loading" anything. But I guess script is named like that...

> partially because I'm lazy

Been there :)

> Should it be a separate rst or just within options like ytdl_hook is?

Depends how much text it is. If it gets too long I would put it in separate rst. I mention the documentation, because while script is relatively basic in principle, the additional options are not clear, unless you really know how it works.

```
disabled=no
images=no
videos=yes
audio=yes
additional_image_exts=list,of,ext
additional_video_exts=list,of,ext
additional_audio_exts=list,of,ext
ignore_hidden=yes
same_type=yes
directory_mode=recursive
ignore_patterns=^~,^bak-,%.bak$
```
there are additional extensions, but also default ones, which cannot be controlled, same_type which select only specific type of files, ignore patters, directory mode. Recently there were issue where there was misunderstanding how same_type works and in general it is not clear unless you read the code.